src/main/java/com/github/manolo8/darkbot/core/objects/facades/QuestProxy.java

```diff
@@ -0,0 +1,336 @@
+package com.github.manolo8.darkbot.core.objects.facades;
+
+import com.github.manolo8.darkbot.core.itf.Updatable;
+import com.github.manolo8.darkbot.core.objects.swf.FlashList;
+import com.github.manolo8.darkbot.core.objects.swf.ObjArray;
+import com.github.manolo8.darkbot.core.utils.ByteUtils;
+
+import eu.darkbot.api.managers.QuestAPI;
+
+import static com.github.manolo8.darkbot.Main.API;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jetbrains.annotations.Nullable;
+
+public class QuestProxy extends Updatable implements QuestAPI {
+    private @Nullable QuestProxy.Quest currentQuest;
+
+    private final List<QuestProxy.QuestListItem> questItems = new ArrayList<>();
+
+    private final ObjArray questItemsArr = ObjArray.ofArrObj(true);
+
+    private @Nullable QuestProxy.Quest questGiverSelected;
+    private @Nullable QuestProxy.QuestListItem questInfoGiverSelected;
+
+    private boolean visibleQuestGiver;
+    private int tabSelected;
+
+    @Override
+    public void update() {
+        if (address == 0) {
+            return;
+        }
+
+        long questClass = API.readMemoryPtr(address + 0x98);
+
+        long currentQuestAddr = API.readMemoryPtr(questClass + 0x28);
+
+        if (currentQuestAddr == 0) {
+            this.currentQuest = null;
+        } else {
+            if (this.currentQuest == null) {
+                this.currentQuest = new Quest();
+            }
+
+            this.currentQuest.update(currentQuestAddr);
+        }
```

Comment:
Instead of checking nulls & address here, you can make `currentQuest, questInfoGiverSelected, questGiverSelected` final, and only in the getters check the address `return currentQuest.address == 0 ? null : currentQuest`.
I see that you already return on address 0 in update() methods of those classess

Reply:
done